begrun.c

```diff
@@ -196,3 +202,105 @@ void close_outputfiles(void)
 }
 
 
+/*! Computes conversion factors between internal code units and the
+ *  cgs-system.
+ */
+static void
+set_units(void)
+{
```

Comment:
Can you set TimeBegin here as well, since it is read in petaio_read_header?
It might also be worth reading OmegaBaryon from the IC file directly.